Add_Keyboard_Support/DriverImplementation.md

```diff
@@ -0,0 +1,217 @@
+# Handling scancodes
+
+Handling the keyboard interrupt was pretty simple, we just basically needed to add an irq to the APIC and read the scancode. But at the end what we got is just a number, and we want to eventually translated it into a character. 
+
+In this section we will see see how to build the Keyboard driver, and what are its responsibilities. 
+
+First of all the driver is not responsible of translating the scancode into printable characters (well sometime on early stages of development it does it too...)
+
+What it cares is only to keep track of the various keyboard events and make them available to any service that needs them.
+
+As already mentioned there are 3 different scancode sets, we will focus on just one (the set 1, since most of the keyboard even if using a different set will have the controller that automatically translates the scancode to that), by the way we will try to implement a generic way to translate, so when eventually a new set needs to be added the changes needed will be very little. 
```

Comment:
the set 1 -> set 1
"will have the controller" what controller? the keyboard controller? I would say "will have the ps2 controller automatically translate set 2 (the default on power-on) to set 1."